lib/routes/routeBackbeat.js

```diff
@@ -304,12 +304,14 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
             // use original data locations
             omVal.location = objMd.location;
         }
-        // specify both 'versioning' and 'versionId' to create a "new"
-        // version (updating master as well) but with specified
-        // versionId
+        const { isNFS, status } =  omVal.replicationInfo;
+        // Specify both 'versioning' and 'versionId' to create a "new" version
+        // (updating master as well) but with specified versionId. If the object
+        // is from a source bucket without versioning (i.e. NFS), then we want
+        // to create a version for the replica object even though none was
+        // provided in the object metadata value.
         const options = {
-            versioning: omVal.versionId !== undefined ||
-                omVal.replicationInfo.isNFS === true,
+            versioning: isNFS !== true || status === 'REPLICA',
```

Comment:
minor comment: we should avoid using comparison operators on booleans, unless there's a very specific reason to do it: use `!isNfs` instead.